Coordinate.cpp

```diff
@@ -1,11 +1,78 @@
 #include "Coordinate.h"
+#include <cmath>
 
 Coordinate::Coordinate() : x(0), y(0) {
 }
 
 Coordinate::Coordinate(int x, int y) : x(x), y(y) {
 }
 
+Coordinate::Coordinate(Direction d)
```

Comment:
I'm not sure this is the best way to do this. logically, a direction is not equivalent to a Coordinate, because a Coordinate corresponds to an actual position. Likewise, adding two coordinates doesn't make too much sense.

A distance and a direction is a `DirectionVector`.